src/Sulu/Bundle/AdminBundle/Admin/Routing/ToolbarAction.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Admin\Routing;
+
+use JMS\Serializer\Annotation\Groups;
+
+class ToolbarAction
+{
+    /**
+     * @var string
+     * @Groups({"frontend"})
+     */
+    private $type;
+
+    /**
+     * @var array
+     * @Groups({"frontend"})
+     */
+    private $options;
+
+    public function __construct(string $type, array $options = [])
+    {
+        $this->type = $type;
+        $this->options = $options;
```

Comment:
'type'  can not be used as option or? should we do something like this here or on another place

```
if (array_key_exists('type', $options)) {
    throw new \DomainException(...);
}
```

Reply:
No, that was the entire point of this operation: The `type` we have introduced here is not part of the options, so you can have another `type` in the options.

Reply:
👍